### PR TITLE
Dataset class

### PR DESCRIPTION
**Pull request by kerventkm**

### CODE REVIEW

**Review by v-goncharenko — COMMENTED** [review]
1. Move `datasests.py` to package directory

File: datasets.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import torch
+import moabb.datasets
```
Comment (v-goncharenko) [inline datasets.py]: Add `torch` and `moabb` to dependencies list

File: datasets.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    def __getitem__(self, item_idx: int):
+        current_sample = self.m_data[item_idx]        
+        return torch.tensor(current_sample)  
+
+    def __len__(self) -> int:
+        return len(self.m_data)
+
+
+class Dataset:
```
Comment (v-goncharenko) [inline datasets.py]: Classes have the same name, shoud be different or remove one

File: datasets.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+m_dataset = moabb.datasets.bi2013a(
+    NonAdaptive=True,
+    Adaptive=True,
+    Training=True,
+    Online=True,
+)
+
+m_dataset.download()
+
+m_data = m_dataset.get_data()
```
Comment (v-goncharenko) [inline datasets.py]: Don't perform heavy operations on top level of any module, move this code to class constructor

File: datasets.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                markers.append(data[-1])
+        raw_dataset.append((eegs, markers))        
+        return raw_dataset
+    
+    def channels(self):
+        return self.m_data[1]['session_1']['run_1'].ch_names[:-1]        
+
+    def __getitem__(self, item_idx: int):
+        current_sample = self.m_data[item_idx]        
+        return torch.tensor(current_sample)
```
Comment (v-goncharenko) [inline datasets.py]: This sould return dict like it was in the first version

File: datasets.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            eegs, markers = [], []
+            for item, run in sorted(sessions['session_1'].items()):
+                data = run.get_data()
+                eegs.append(data[:-1])
+                markers.append(data[-1])
+        raw_dataset.append((eegs, markers))        
+        return raw_dataset
+    
+    def channels(self):
+        return self.m_data[1]['session_1']['run_1'].ch_names[:-1]
```
Comment (v-goncharenko) [inline datasets.py]: You have to install pre-commit on your local repo

**Comment by kerventkm** [comment]
> 1. Move `datasests.py` to package directory

Done

**Review by v-goncharenko — APPROVED** [review]

**Pull request merged**